### PR TITLE
.Net: {Azure}OpenAI Connectors Update to 2.1.0-beta.1

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

Update to the latest pre-release of OpenAI and Azure OpenAI pre-release packages following the new guidelines described in the ADR.

Moving forward connectors that target pre-releases will also be tagged as pre-release as well as the `Microsoft.SemanticKernel` meta package (that targets pre-release version of OpenAI and AzureOpenAI connectors.

THis update enable usage of the Realtime APIs provided by the underlying SDKs allowing them to be used for breaking glass scenarios.

### CODE REVIEW

File: dotnet/samples/Concepts/Agents/ComplexChat_NestedShopper.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 using OpenAI.Chat;
 using Resources;
 
+using ChatMessageContent = Microsoft.SemanticKernel.ChatMessageContent;
```
Comment (crickman) [inline dotnet/samples/Concepts/Agents/ComplexChat_NestedShopper.cs:10]: I see what's happened here.  Given teh context, might this be more clear/targeted?

```suggestion
using ChatResponseFormat = OpenAI.Chat.ChatResponseFormat;
```

Note: This will result in the removal of the `OpenAI.Chat` namespace...but the PR won't allow the suggestion to extend to un-edited lines:

```c#
using Microsoft.SemanticKernel;
using Microsoft.SemanticKernel.Agents;
using Microsoft.SemanticKernel.Agents.Chat;
using Microsoft.SemanticKernel.ChatCompletion;
using Microsoft.SemanticKernel.Connectors.OpenAI;
using Resources;

using ChatResponseFormat = OpenAI.Chat.ChatResponseFormat;
```

**Review by crickman — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAIPromptExecutionSettings.cs [hunk 2]
```diff
@@ -101,8 +101,8 @@ public static AzureOpenAIPromptExecutionSettings FromExecutionSettingsWithData(P
     }
 
     #region private ================================================================================
-
-    private AzureChatDataSource? _azureChatDataSource;
+    [Experimental("SKEXP0010")]
+    private AzureSearchChatDataSource? _azureChatDataSource;
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAIPromptExecutionSettings.cs:105]: Is `Experimental` flag required on private variable?

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.AudioToText.cs [hunk 3]
```diff
@@ -62,15 +62,22 @@ private static AudioTranscriptionOptions AudioOptionsFromExecutionSettings(OpenA
...
+        }
+
+        AudioTranscriptionFormat result;
+        switch (responseFormat)
+        {
+            case "json": result = AudioTranscriptionFormat.Simple; break;
+            case "verbose_json": result = AudioTranscriptionFormat.Verbose; break;
+            case "vtt": result = AudioTranscriptionFormat.Vtt; break;
+            case "srt": result = AudioTranscriptionFormat.Srt; break;
+            default: throw new NotSupportedException($"The audio transcription format '{responseFormat}' is not supported.");
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.AudioToText.cs:77]: Is this change required? The same is related to other `switch/case` changes in this PR. If it's not required, let's try to minimize the number of changes needed for package upgrade.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAIPromptExecutionSettings.cs (hunk 2 above)
Comment (RogerBarreto) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAIPromptExecutionSettings.cs:105]: If I don't do that I need to add a pragma for the Azure Experimental in here.

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.AudioToText.cs (hunk 3 above)
Comment (RogerBarreto) [reply dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.AudioToText.cs:77]: Unfortunately this change is required, as their constructor changes breaks the usage of switch expressions `ArgumentNullException`.

**Review by dmytrostruk — APPROVED** [review]
